src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2194 @@
+
+            Assert.Equal("Int32", typeSpeculation.Type.Name);
+        }
+
+        [Fact]
+        public void BlockBodyAndExpressionBody_14()
+        {
+            var comp = CreateCompilation(new[] { @"
+public class C
+{
+    static int P1 { get; set; }
+    int P2
+    {
+        init { P1 = 1; } => P1 = 1;
+    }
+}
+", IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+
+            comp.VerifyDiagnostics(
+                // (7,9): error CS8057: Block bodies and expression bodies cannot both be provided.
+                //         init { P1 = 1; } => P1 = 1;
+                Diagnostic(ErrorCode.ERR_BlockBodyAndExpressionBody, "init { P1 = 1; } => P1 = 1;").WithLocation(7, 9)
+                );
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var nodes = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>();
+            Assert.Equal(2, nodes.Count());
+
+            foreach (var assign in nodes)
+            {
+                var node = assign.Left;
+                Assert.Equal("P1", node.ToString());
+                Assert.Equal("System.Int32 C.P1 { get; set; }", model.GetSymbolInfo(node).Symbol.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void ModReqOnAccessorParameter()
+        {
+            string il = @"
+.class public auto ansi beforefieldinit C extends System.Object
+{
+    .method public hidebysig specialname newslot virtual instance void set_Property ( int32 modreq(System.Runtime.CompilerServices.IsExternalInit) 'value' ) cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+
+    .property instance int32 Property()
+    {
+        .set instance void C::set_Property(int32 modreq(System.Runtime.CompilerServices.IsExternalInit))
+    }
+}
+
+.class public auto ansi sealed beforefieldinit System.Runtime.CompilerServices.IsExternalInit extends System.Object
+{
+    .method public hidebysig specialname rtspecialname instance void .ctor () cil managed
+    {
+        IL_0000: ldnull
+        IL_0001: throw
+    }
+}
+";
+
+            string source = @"
+public class Derived : C
+{
+    public override int Property { set { throw null; } }
+}
+public class Derived2 : C
+{
+    public override int Property { init { throw null; } }
+}
+";
+
+            var reference = CreateMetadataReferenceFromIlSource(il);
+            var comp = CreateCompilation(source, references: new[] { reference }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,25): error CS0569: 'Derived.Property': cannot override 'C.Property' because it is not supported by the language
```

Comment:
>// (4,25): error CS0569: 'Derived.Property': cannot override 'C.Property' because it is not supported by the language [](start = 16, length = 117)

Does modreq have anything to do with this error? C# doesn't support properties with parameters, only indexers.

Reply:
Yes. Without the modreq in IL, this test produces the following diagnostics. The above IL is that for a regular property, with modreq added on the setter parameter instead of return value.

```
                    // (8,25): error CS8803: 'Derived2.Property' must match by init-only of overridden member 'C.Property'
                    //     public override int Property { init { throw null; } }
                    Diagnostic(ErrorCode.ERR_CantChangeInitOnlyOnOverride, "Property").WithArguments("Derived2.Property", "C.Property").WithLocation(8, 25)
```

---
In reply to: [413380794](https://github.com/dotnet/roslyn/pull/43275#discussion_r413380794) [](ancestors = 413380794)

Reply:
> Yes. Without the modreq in IL, this test produces the following diagnostics. The above IL is that for a regular property, with modreq added on the setter parameter instead of return value.

Rather than relying solely on this diagnostics difference (one error vs. another not quite specific error), please add an assert that the setter has a use-site error.
Also, it would be good to test scenario when a success case becomes an error after addition of a modreq on a parameter type. For example, if it is done on a regular method, an attempt to call it should become an error, and the method should have a use-site error.  

---
In reply to: [413412858](https://github.com/dotnet/roslyn/pull/43275#discussion_r413412858) [](ancestors = 413412858,413380794)

Reply:
Since we allow init-only indexers it would be good to have a test with an indexer (or convert this test to an indexer).

---
In reply to: [415197789](https://github.com/dotnet/roslyn/pull/43275#discussion_r415197789) [](ancestors = 415197789,413412858,413380794)

Reply:
Should also add tests  for case when there is a modifier on the return type in the property signature. We shouldn't be able to consume such property. Should assert symbols for this situation as well.

---
In reply to: [415210284](https://github.com/dotnet/roslyn/pull/43275#discussion_r415210284) [](ancestors = 415210284,415197789,413412858,413380794)

Reply:
Should also add tests  for case when there is a modifier on the parameter type in the property signature.

---
In reply to: [415217269](https://github.com/dotnet/roslyn/pull/43275#discussion_r415217269) [](ancestors = 415217269,415210284,415197789,413412858,413380794)